src/pudl_rmi/coordinate.py

```diff
@@ -0,0 +1,247 @@
+            clobber_ppl (boolean): Generate and cache a new interim
+                output of the EIA plant part list and generate a new version of
+                the depreciaiton to FERC1 output. Default is False.
+            clobber_de (boolean): Generate and cache a new interim output of
+                the connection between EIA and depreciation data and generate
+                a new version of the depreciaiton to FERC1 output. Default is
+                False.
+            clobber_fe (boolean): Generate and cache a new interim output of
+                the connection between FERC and EIA and generate a new version
+                of the depreciaiton to FERC1 output. Default is False.
+
+        Returns:
+            pandas.DataFrame: depreciation study data connected to FERC1 data
+            which has been scaled down or aggregated to the level of reporting
+            in the depreciaiton studies.
+        """
+        # if any of the clobbers are on, we want to regenerate the main output
+        clobber_any = any([clobber, clobber_ppl, clobber_de, clobber_fe])
+        if not pudl_rmi.FILE_PATH_DEPRISH_FERC1.is_file() or clobber_any:
+            logger.info(
+                "Deprish to FERC1 granular connection not found at "
+                f"{pudl_rmi.FILE_PATH_DEPRISH_FERC1}. Generating a new output."
+            )
+            connects_deprish_ferc1 = pudl_rmi.connect_deprish_to_ferc1.execute(
+                plant_parts_eia=self.get_plant_part_list(clobber=clobber_ppl),
+                deprish_eia=self.get_deprish_to_eia(clobber=clobber_de),
+                ferc1_to_eia=self.get_ferc1_to_eia(clobber=clobber_fe),
+                clobber=clobber
+            )
+            # export
+            connects_deprish_ferc1.to_pickle(pudl_rmi.FILE_PATH_DEPRISH_FERC1)
+
+        else:
+            logger.info(
+                "Reading the depreciation to FERC1 connection from "
+                f"{pudl_rmi.FILE_PATH_DEPRISH_FERC1}"
+            )
+            connects_deprish_ferc1 = pd.read_pickle(
+                pudl_rmi.FILE_PATH_DEPRISH_FERC1
+            )
+        return connects_deprish_ferc1
+
+    def get_all(self, clobber_all=False):
+        """
+        Gotta catch em all. Get all of the RMI outputs.
+
+        Get or regenerate all of the RMI outputs.
+
+        Args:
+            clobber_all (boolean): Deafult is False, which will grab the
+                outputs if they already exist, or generate them if they don't
+                exist. True will re-generate the outputs whether they exist on
+                disk. Re-generating everything will take ~15 minutes.
+        """
+        ppl = self.get_plant_part_list(clobber=clobber_all)
+        d = self.get_deprish(clobber=clobber_all)
+        de = self.get_deprish_to_eia(clobber=clobber_all)
+        fe = self.get_ferc1_to_eia(clobber=clobber_all)
+        df = self.get_deprish_to_ferc1(clobber=clobber_all)
+        return ppl, d, de, fe, df
```

Comment:
Returning sequences of things (especially 5 of them) that don't have any inherent sequence can be hard to keep track of At the very least the docstring needs to explain what order everything is going to come back in. It would be clearer to use a dictionary.  Or maybe better yet don't return anything at all -- just set the attributes of the object, which can be accessed by their names after being calculated.

Reply:
i went with adding more docs here.